miden/src/cli/prove.rs

```diff
@@ -58,8 +62,20 @@ impl ProveCmd {
         let exec_options =
             ExecutionOptions::new(Some(self.max_cycles), self.expected_cycles, self.tracing)?;
         Ok(match self.security.as_str() {
-            "96bits" => ProvingOptions::with_96_bit_security(self.recursive),
-            "128bits" => ProvingOptions::with_128_bit_security(self.recursive),
+            "96bits" => {
+                if !self.recursive && self.rpx {
+                    ProvingOptions::with_96_bit_security_rpx()
+                } else {
+                    ProvingOptions::with_96_bit_security(self.recursive)
+                }
+            }
+            "128bits" => {
+                if self.recursive && self.rpx {
```

Comment:
Should this be `!self.recursive && self.rpx` similar to line 66 above?

Reply:
removed the !self.recursive part, since its redundant now